gms/impl/src/main/resources/mlModelESSearchQueryTemplate.json

```diff
@@ -0,0 +1,67 @@
+{
+  "function_score": {
+    "query": {
+      "bool": {
+        "should": [
+          {
+            "query_string": {
+              "query": "$INPUT",
+              "analyzer": "whitespace_lowercase",
+              "fields": [
+                "evaluationDatasets^0.065",
+                "trainingDatasets^0.075",
+                "description^0.035",
+                "platform^0.055",
+                "type^0.01"
+              ],
+              "default_operator": "OR"
```

Comment:
could you explain why `default_operator` is OR for some fields and `AND` for others?

Reply:
Usage of AND operator for name is to be aligned with Dataset, for the other fields we felt OR was more appropriate. For example if you're searching for a something with training datasets "trainingDatasetOne trainingDatasetTwo" it is more intuitive to me to return the models with either one from the raw query and use filters to limit results. I can see use cases for either approach though so I'm open to changing it to align on one operator if you think that would be better.

Reply:
Makes sense. However since we have many fields in the same `query_string` clause, the query will be matched across all these fields. So for your example, let's say the input query is `trainingDatasetOne trainingDatasetTwo` and default_operator is chosen as `OR` then the query executed (ignoring the boosts) is essentially `(evaluationDatasets: trainingDatasetOne OR trainingDatasets:trainingDatasetOne OR description:trainingDatasetOne OR platform:trainingDatasetOne OR type:trainingDatasetOne) OR (evaluationDatasets: trainingDatasetTwo OR trainingDatasets:trainingDatasetTwo OR description:trainingDatasetTwo OR platform:trainingDatasetTwo OR type:trainingDatasetTwo)`. Is that what you intended? If the intention was to apply the `OR` logic on training and evaluation datasets ONLY, then I suggest we move the rest of the fields (i.e. description, platform, type) to a separate `query_string` clause with `AND` as default_operator. If not, the we don't need any changes.

Reply:
Yeah, from my perspective if the dataset name shows up in any of the fields then it should show up in the result set. For example if the description of it was "Takes training data from trainingDatasetOne..." then I would expect it to show up even if the TrainingData aspect had not been properly populated by some pipeline.

I'm not experienced enough on ES to be able to tell if that is a big performance concern though so if it is I'm totally open to breaking it apart into multiple query_strings if that makes more sense from typical usage and performance.

 I am curious how it would operate with multiple query_strings though since there's only one input string from the search methods, could you elaborate on how it would operate differently?

Reply:
It looks good to me now. My suggestion is to use one operator as default operator query string, either OR or AND, to reduce the possible confusions from users.  With query_string,  user can always specify and overwrite the default operator. Full syntax can be found here: 
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#query-string-syntax

Reply:
I can consolidate them, in my mind it makes sense to distinguish between the default behavior for searching by name vs searching for other fields, but I can see how that might be unintuitive for different fields to behave differently by default.